src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -190,6 +198,66 @@ public Response updateByDatasetUpdate(
     }
   }
 
+  /**
+   * This endpoint updates the dataset.
+   */
+  @PATCH
+  @Consumes({MediaType.APPLICATION_JSON})
+  @Produces({MediaType.APPLICATION_JSON})
+  @Path("/{datasetId}")
+  @RolesAllowed({ADMIN, CHAIRPERSON, DATASUBMITTER})
+  public Response patchByDatasetUpdate(@Auth AuthUser authUser,
+      @PathParam("datasetId") Integer datasetId, String json) {
+    try {
+      if (json == null || json.isEmpty()) {
+        throw new BadRequestException("Dataset Patch is required");
+      }
+      Gson gson = GsonUtil.getInstance();
+      DatasetPatch patch;
+      try {
+        patch = gson.fromJson(json, DatasetPatch.class);
+      } catch (Exception e) {
+        throw new BadRequestException("Unable to parse dataset patch: " + json);
+      }
```

Comment:
We can't rely only on swagger for validation, we have to have it here as well. Nate hits a lot of the dataset APIs via scripts.

Reply:
The main difference here is that TDR relies on openapi (swagger) code generation, which converts the openapi yaml into java code. The generated code enforces validation (and also generates the equivalent `@PATCH`, `@Path()`, etc annotations too). Since consent doesn't use openapi codegen, it must manually implement these features of the swagger yaml.

I think the biggest takeaway here is that, with this approach, there is no enforcement that the openapi yaml matches the java implementation. Because of this, most (all?) other Java terra services use openapi codegen to avoid code disagreement and duplication.

Reply:
> I think the biggest takeaway here is that, with this approach, there is no enforcement that the openapi yaml matches the java implementation. Because of this, most (all?) other Java terra services use openapi codegen to avoid code disagreement and duplication.

This is correct, and essentially an artifact of history. Swagger codegen tools were not around when Consent/Ontology were set up and it's never been a priority for us to do a conversion.